internal/build/phase_factory.go

```diff
@@ -0,0 +1,40 @@
+package build
+
+import (
+	"fmt"
+)
+
+type DefaultPhaseFactory struct {
+	lifecycle *Lifecycle
+}
+
+func NewDefaultPhaseFactory(lifecycle *Lifecycle) *DefaultPhaseFactory {
+	return &DefaultPhaseFactory{lifecycle: lifecycle}
+}
+
+func (m *DefaultPhaseFactory) New(name string, provider *PhaseConfigProvider) (RunnerCleaner, error) {
+	err := provider.Update(
+		WithLifecycle(m.lifecycle),
+		WithBinds([]string{
+			fmt.Sprintf("%s:%s", m.lifecycle.LayersVolume, layersDir),
+			fmt.Sprintf("%s:%s", m.lifecycle.AppVolume, appDir),
+		}...),
+	)
```

Comment:
Upon reflection I think this belongs in the constructor for the provider.

Reply:
I am also going to break up `WithLifecycle` as right now it is doing both "image" and "proxy" things.